core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -48,7 +48,10 @@
     this.schema = new Schema(
         Types.NestedField.required(1, "partition", table.spec().partitionType()),
         Types.NestedField.required(2, "record_count", Types.LongType.get()),
-        Types.NestedField.required(3, "file_count", Types.IntegerType.get())
+        Types.NestedField.required(3, "file_count", Types.IntegerType.get()),
+        Types.NestedField.required(4, "delete_file_count", Types.IntegerType.get()),
+        Types.NestedField.required(5, "equality_delete_count", Types.LongType.get()),
```

Comment:
There might be a little confusion in the name (I thought it means the equality_delete_files and position_delete_files count at first).  Maybe we can change to "equality_delete_record_count" and "position_delete_record_count".

Not sure what others think

Reply:
I agree. It would be better to use `file` and `record` to clarify what each count means.

It would also be nice to rename `file_count`, but that probably isn't a good idea because it would break existing queries. Instead, we should add field docs to explain.

Reply:
I agree the name should be clear. But I use the old name is to keep consistent with [SnapshotSummary.](https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/SnapshotSummary.java#L30)

Reply:
I'd like to add file_size to the schema, do you think it is necessary? @rdblue

Reply:
file_size, equality_delete_file_size and position_delete_file_size.

Reply:
The names still do not seem consistent to me ("delete_file" and "file" but "equality_delete" and "position_delete").  But yes good point this is following  SnapshotSummary, though I think this schema is already quite different than that.  Will leave to @rdblue or others to see what they think.

Reply:
Not sure if Ryan had any comment, so I still think we should rename it to "positional/equality_delete_record_count".  I dont see any point necessarily in staying with snapshot-summary if that one is less clear.